numpy/core/records.py

```diff
@@ -772,8 +772,53 @@ def fromrecords(recList, dtype=None, shape=None, formats=None, names=None,
 
 def fromstring(datastring, dtype=None, shape=None, offset=0, formats=None,
                names=None, titles=None, aligned=False, byteorder=None):
-    """Create a (read-only) record array from binary data contained in
-    a string"""
+    """Create a record array from binary data
+
+    Note that despite the name of this function it does not accept `str`
+    instances
+
+    Parameters
+    ----------
+    datastring : bytes-like
+        Binary data contained in a string
```

Comment:
Not a fan of the word "string" here. Maybe
```suggestion
        Buffer of binary data
```